src/redux/offerSlice.js

```diff
@@ -0,0 +1,43 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getOfferDetails } from '../api';
+
+const initialState = {
+  offer: null,
+  loading: false,
+  error: null,
+  isOfferFree: false
+};
+
+export const fetchOffer = createAsyncThunk(
+  'offer/fetchOffer',
+  async orderId => {
```

Comment:
offerId

Reply:
I don't see this changes -> you are fetching here a offer, so you should pass a offerId
`async offerId => {`
and 
`const { responseData } = await getOfferDetails(offerId);`